qa/common/src/main/java/org/eclipse/kapua/qa/common/TestJAXBClassProvider.java

```diff
@@ -83,121 +96,117 @@
 import org.eclipse.kapua.service.job.Job;
 import org.eclipse.kapua.service.job.JobListResult;
 import org.eclipse.kapua.service.job.JobXmlRegistry;
-import org.eclipse.persistence.jaxb.JAXBContextFactory;
-import org.eclipse.persistence.jaxb.MarshallerProperties;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-import javax.xml.bind.JAXBContext;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.Arrays;
+import java.util.Collection;
 
 /**
  * JAXB context provided for proper (un)marshalling of interface annotated classes.
  * This particular implementation is used only in unit and integration tests.
  * <p>
  * Application and interfaces have their own implementation of provider.
  */
-public class TestJAXBContextProvider implements JAXBContextProvider {
-
-    private static final Logger LOG = LoggerFactory.getLogger(TestJAXBContextProvider.class);
-
-    private JAXBContext context;
-
+public class TestJAXBClassProvider implements ClassProvider {
     @Override
```

Comment:
Separation of concerns. This class now just provides the classes to be considered for serialization in tests.

Note: this was left just because the mess of module dependencies in tests prevents to perform class path scan, as it should be done in a saner environment.